src/spi.rs

```diff
@@ -23,19 +23,50 @@ pub enum Error {
     _Extensible,
 }
 
-pub trait Pins<SPI> {
-    const REMAP: bool;
-}
+// FIXME these should be "closed" traits
+/// SCK pin -- DO NOT IMPLEMENT THIS TRAIT
+pub trait SckPin<SPI> {}
 
-impl Pins<SPI1>
-    for (
-        PA5<Alternate<AF5, Input<Floating>>>,
-        PA6<Alternate<AF5, Input<Floating>>>,
-        PA7<Alternate<AF5, Input<Floating>>>,
-    )
-{
-    const REMAP: bool = false; // TODO REMAP
-}
+/// MISO pin -- DO NOT IMPLEMENT THIS TRAIT
+pub trait MisoPin<SPI> {}
+
+/// MOSI pin -- DO NOT IMPLEMENT THIS TRAIT
+pub trait MosiPin<SPI> {}
+
+impl SckPin<SPI1> for gpio::gpioa::PA5<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI1> for gpio::gpioa::PA6<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI1> for gpio::gpioa::PA7<Alternate<AF5, Input<Floating>>> {}
+impl SckPin<SPI1> for gpio::gpiob::PB3<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI1> for gpio::gpiob::PB4<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI1> for gpio::gpiob::PB5<Alternate<AF5, Input<Floating>>> {}
+impl SckPin<SPI1> for gpio::gpioe::PE13<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI1> for gpio::gpioe::PE14<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI1> for gpio::gpioe::PE15<Alternate<AF5, Input<Floating>>> {}
+// missing gpiog
+//impl SckPin<SPI1> for gpio::gpiog::PG2<Alternate<AF5, Input<Floating>>> {}
+//impl MisoPin<SPI1> for gpio::gpiog::PG3<Alternate<AF5, Input<Floating>>> {}
+//impl MosiPin<SPI1> for gpio::gpiog::PG4<Alternate<AF5, Input<Floating>>> {}
+
+impl SckPin<SPI2> for gpio::gpiob::PB13<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI2> for gpio::gpiob::PB14<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI2> for gpio::gpiob::PB15<Alternate<AF5, Input<Floating>>> {}
+impl SckPin<SPI2> for gpio::gpiob::PB10<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI2> for gpio::gpioc::PC2<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI2> for gpio::gpioc::PC3<Alternate<AF5, Input<Floating>>> {}
+impl SckPin<SPI2> for gpio::gpiod::PD1<Alternate<AF5, Input<Floating>>> {}
+impl MisoPin<SPI2> for gpio::gpiod::PD3<Alternate<AF5, Input<Floating>>> {}
+impl MosiPin<SPI2> for gpio::gpiod::PD4<Alternate<AF5, Input<Floating>>> {}
+
+impl SckPin<SPI3> for gpio::gpiob::PB3<Alternate<AF6, Input<Floating>>> {}
+impl MisoPin<SPI3> for gpio::gpiob::PB4<Alternate<AF6, Input<Floating>>> {}
+impl MosiPin<SPI3> for gpio::gpiob::PB5<Alternate<AF6, Input<Floating>>> {}
+impl SckPin<SPI3> for gpio::gpioc::PC10<Alternate<AF6, Input<Floating>>> {}
+impl MisoPin<SPI3> for gpio::gpioc::PC11<Alternate<AF6, Input<Floating>>> {}
+impl MosiPin<SPI3> for gpio::gpioc::PC12<Alternate<AF6, Input<Floating>>> {}
+// missing gpiog
```

Comment:
If these are missing, feel free to add a PR adding them, else remove these for now.

Reply:
I'd like to, but I don't quite understand where the arguments for the macro are coming from. Specifically, `gpio!` gets passed `gpioa` twice (the second and third argument) and for `gpiod` it gets passed `gpioc` as third argument. Do you know what this argument is based on?

Reply:
I've figured it out now.